cloud/pkg/admissioncontroller/mutate_offlinemigration.go

```diff
@@ -0,0 +1,91 @@
+package admissioncontroller
+
+import (
+	"encoding/json"
+	"net/http"
+
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+type patchMapValue struct {
+	Op    string              `json:"op"`
+	Path  string              `json:"path"`
+	Value []map[string]string `json:"value,omitempty"`
+}
+
+func mutateOfflineMigration(review admissionv1beta1.AdmissionReview) *admissionv1beta1.AdmissionResponse {
+	reviewResponse := admissionv1beta1.AdmissionResponse{
+		Allowed: true,
+	}
+	var payload []patchMapValue
+	switch review.Request.Kind.Kind {
+	case "Deployment":
+		var deployment v1.Deployment
+		if err := json.Unmarshal(review.Request.Object.Raw, &deployment); err != nil {
+			klog.Errorf("Could not unmarshal raw object: %v", err)
+			return toAdmissionResponse(err)
+		}
+
+		payload = generatePatch(deployment.Spec.Template.Spec.Tolerations)
+
+	case "StatefulSet":
+		var statefulset v1.StatefulSet
+		if err := json.Unmarshal(review.Request.Object.Raw, &statefulset); err != nil {
+			klog.Errorf("Could not unmarshal raw object: %v", err)
+			return toAdmissionResponse(err)
+		}
+		payload = generatePatch(statefulset.Spec.Template.Spec.Tolerations)
+	case "DaemonSet":
+		var daemonset v1.DaemonSet
+		if err := json.Unmarshal(review.Request.Object.Raw, &daemonset); err != nil {
+			klog.Errorf("Could not unmarshal raw object: %v", err)
+			return toAdmissionResponse(err)
+		}
+		payload = generatePatch(daemonset.Spec.Template.Spec.Tolerations)
+	}
+
+	if len(payload) == 0 {
+		return &reviewResponse
+	}
+	patch, err := json.Marshal(payload)
+	if err != nil {
+		return toAdmissionResponse(err)
+	}
+
+	reviewResponse.Patch = patch
+	pt := admissionv1beta1.PatchTypeJSONPatch
+	reviewResponse.PatchType = &pt
+	return &reviewResponse
+}
+
+func generatePatch(tolerations []corev1.Toleration) []patchMapValue {
+	patch := []patchMapValue{{
+		Op:   "add",
+		Path: "/spec/template/spec/tolerations",
+		Value: []map[string]string{{
+			"key":      corev1.TaintNodeUnreachable,
+			"operator": "Exists",
+			"effect":   "NoExecute",
+		}},
+	}}
+	if len(tolerations) > 0 {
+		for _, toleration := range tolerations {
+			if toleration.Key == corev1.TaintNodeUnreachable {
+				if toleration.Effect == "NoExecute" &&
+					toleration.Operator == "Exists" {
+					return nil
+				}
+				patch[0].Op = "replace"
```

Comment:
Now K8s will add toleration to pod as below:
```
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
```
Do we need to delete the deafultTimeout here?

Reply:
I think so, or it would restart when timeout.

Reply:
So please fix it.